RevenueCatUI/Templates/Components/TemplateComponentsView.swift

```diff
@@ -51,14 +50,34 @@ public struct TemplateComponentsView: View {
         .edgesIgnoringSafeArea(.top)
     }
 
-}
+    static func chooseLocalization(
+        for componentsData: PaywallComponentsData
+    ) -> (locale: Locale, localizedStrings: PaywallComponent.LocalizationDictionary) {
 
-func getLocalization(_ locale: Locale, _ displayString: DisplayString) -> String {
-    if let found = displayString.value[locale.identifier] {
-        return found
-    }
+        guard !componentsData.componentsLocalizations.isEmpty else {
+            Logger.error(Strings.paywall_contains_no_localization_data)
+            return (Locale.current, PaywallComponent.LocalizationDictionary())
+        }
+
+        // STEP 1: Get available paywall locales
+        let paywallLocales = componentsData.componentsLocalizations.keys.map { Locale(identifier: $0) }
 
-    return displayString.value.values.first!
+        let fallbackLocale = Locale(identifier: componentsData.defaultLocale)
```

Comment:
Nit: Can we call this `defaultLocale` 😇

Reply:
We can - I chose fallback because I felt it is a more precise description of its function here, where "default" is what is chosen if you make no choice, and "fallback" is what you get when your choice is unavailable. However "default" is the user-facing term, so it makes sense to keep things consistent.

Reply:
We also used "fallback" as the "first language" if no default was selected in Paywalls V1 so it kind of has a different meaning in my mind too 🫠 

We don't need that same behavior here but I'm already broken haha

Reply:
🫠 
Changed to default!